packages/mui-joy/src/SvgIcon/SvgIcon.tsx

```diff
@@ -0,0 +1,176 @@
+import { unstable_composeClasses as composeClasses } from '@mui/base';
+import { OverridableComponent } from '@mui/types';
+import { unstable_capitalize as capitalize } from '@mui/utils';
+import clsx from 'clsx';
+import PropTypes from 'prop-types';
+import * as React from 'react';
+import styled from '../styles/styled';
+import useThemeProps from '../styles/useThemeProps';
+import { getSvgIconUtilityClass } from './svgIconClasses';
+import { SvgIconProps, SvgIconTypeMap } from './SvgIconProps';
+
+const useUtilityClasses = (ownerState: SvgIconProps) => {
+  const { color, fontSize, classes } = ownerState;
+
+  const slots = {
+    root: [
+      'root',
+      color && `color${capitalize(color)}`,
+      fontSize && `fontSize${capitalize(fontSize)}`,
+    ],
+  };
+
+  return composeClasses(slots, getSvgIconUtilityClass, classes);
+};
+
+const SvgIconRoot = styled('svg', {
+  name: 'MuiSvgIcon',
+  slot: 'Root',
+  overridesResolver: (props, styles) => {
+    const { ownerState } = props;
+
+    return [
+      styles.root,
+      styles[`color${capitalize(ownerState.color)}`],
+      styles[`fontSize${capitalize(ownerState.fontSize)}`],
+    ];
+  },
+})<{ ownerState: SvgIconProps }>(({ theme, ownerState }) => {
+  return [
+    {
+      userSelect: 'none',
+      width: '1em',
+      height: '1em',
+      display: 'inline-block',
+      fill: 'currentColor',
+      flexShrink: 0,
+      transition: 'fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms',
+      fontSize:
+        ownerState.fontSize !== 'inherit' ? theme.vars.fontSize[ownerState.fontSize!] : 'inherit',
+      color:
+        ownerState.color !== 'inherit' && theme.vars.palette[ownerState.color!]
+          ? theme.vars.palette[ownerState.color!].textColor
+          : 'inherit',
+    },
+  ];
+});
+
+const SvgIcon = React.forwardRef(function SvgIcon(inProps, ref) {
+  const props = useThemeProps<typeof inProps & SvgIconProps>({
+    props: inProps,
+    name: 'MuiSvgIcon',
+  });
+
+  const {
+    children,
+    className,
+    color = 'inherit',
+    component = 'svg',
+    fontSize = 'md',
```

Comment:
```suggestion
    fontSize = 'xl',
```

Let's keep `xl` (1.5rem) for now because it is a standard size for icon.